lib/model/model.go

```diff
@@ -1059,10 +1059,10 @@ func (m *model) handleIndex(deviceID protocol.DeviceID, folder string, fs []prot
 
 	if cfg, ok := m.cfg.Folder(folder); !ok || !cfg.SharedWith(deviceID) {
 		l.Infof("%v for unexpected folder ID %q sent from device %q; ensure that the folder exists and that this device is selected under \"Share With\" in the folder configuration.", op, folder, deviceID)
-		return
+		return errors.Wrap(errFolderMissing, folder)
```

Comment:
Just a nitpicky opinion, feel free to dismiss silently: The `folder` addition seems unnecessary to me: The caller knows about the folder and thus could wrap the error with that additional info if it thinks it's useful for whatever context the error is used in. Generally I think errors should be wrapped with information pertinent to the error that wouldn't otherwise be available at the receiver of the error.

Reply:
I can see the merit of that, yet it seems to me like the parameter is often included in the error, compare `open: some-file-name: not found` returned from os.Open. I'm going to ignore your complaint for the moment and meditate on it for a future "harmonize all the errors" pass :)